test/unit/org/apache/cassandra/auth/GrantAndRevokeTest.java

```diff
@@ -357,4 +360,29 @@ public void testGrantedAllTables() throws Throwable
         assertUnauthorizedQuery("User user has no ALTER permission on <table " + table + "> or any of its parents",
                                 "DROP INDEX " + index);
     }
+
+    @Test
+    public void testWarnings() throws Throwable
+    {
+        useSuperUser();
+
+        executeNet("CREATE KEYSPACE revoke_yeah WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}");
+        executeNet("CREATE TABLE revoke_yeah.t1 (id int PRIMARY KEY, val text)");
+        executeNet("CREATE USER '" + user + "' WITH PASSWORD '" + pass + "'");
+
+        ResultSet res = executeNet("REVOKE CREATE ON KEYSPACE revoke_yeah FROM " + user);
+        assertWarningsContain(res.getExecutionInfo().getWarnings(), "Role '" + user + "' was not granted CREATE on <keyspace revoke_yeah>");
```

Comment:
Probably it doesn't matter for this case, but I wonder if we should check that the emitted warnings are exactly the expected ones, so check that there aren't any other unexpected warnings. Perhaps we could have (quite trivial) utility methods for that on `CQLTester`, for example:
```java
protected static void assertWarningsEquals(ResultSet rs, String... messages)
{
    assertWarningsEquals(rs.getExecutionInfo().getWarnings(), messages);
}

protected static void assertWarningsEquals(List<String> warnings, String... messages)
{
    Assert.assertNotNull(warnings);
    Assertions.assertThat(messages).hasSameElementsAs(warnings);
}
```

Reply:
That is a good idea at face value. The problem is that warnings are accumulative in nature. Sometime sbdy is going to add a new warning somewhere and then tests will start to fail and you'll have to fix them. So I am +1 in adding these methods for the occasional test where an exact match can be guaranteed but not here. Wdyt?